File: 03.programmers/lv02/02.Python/n_game.py
dic = {i:hex(i)[2:].upper() for i in range(16)}
def change(number, n):
    global dic
    if number//n == 0: return dic[number]
    else: return change(number//n, n) + dic[number%n]

def solution(n, t, m, p):
    number, index = 0, 0
    answer = ''
    while 1:
        number_str = change(number, n)
        print(number_str)
        for num in number_str:
            if index == p-1: answer += num
            if len(answer)==t: return answer
            index = (index+1) % m
        number += 1

input_n = 2
input_t = 4
input_m = 2
input_p = 1

# input_n = 16
# input_t = 16
# input_m = 2 
# input_p	= 1	
# 02468ACE11111111

print(solution(input_n,input_t,input_m,input_p))

File: 03.programmers/lv02/01.C++/n_game.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string change(int number, int n,string ret){
    char idx = number%n>=10?number%n - 10 +'A':number%n + '0';
    if(number/n==0)return idx + ret;
    else return change(number/n,n,idx + ret);
}
string solution(int n, int t, int m, int p) {
    string answer = "";
    int number = 0, index = 0;
    while(true){
        string number_str = change(number, n, "");
        for(auto num : number_str){
            if(index == p-1)answer +=num;
            if(answer.size() == t)return answer;
            index = (index +1)%m;
        }
        number++;
    }
    return answer;
}
int main(){
    int n = 2;
    int t = 4;
    int m = 2;
    int p = 1;

    cout<<solution(n,t,m,p)<<endl;
    return 0;
}